File: c/queue.c
#include <stdlib.h>
#include <stdio.h>



struct queueNode{
	int data;
	struct queueNode *next;
	struct queueNode *last;
};

void enqueue(struct queueNode **S, int data){
	struct queueNode *newNode = malloc(sizeof(struct queueNode));
	newNode->data = data;
	newNode->next = NULL;
	newNode->last = newNode;
	if(*S == NULL){
		*S = newNode;
	}else{
		(*S)->last->next = newNode;
		(*S)->last = newNode;
	}
}

int dequeue(struct queueNode **S){
	int dequeueData = (*S)->data;
	struct queueNode *last = (*S)->last;
	
	*S = (*S)->next;
	if(*S != NULL)
		(*S)->last = last;

	return dequeueData;
}

int front(struct queueNode **S){
	int dequeueData = (*S)->data;
	return dequeueData;
}


int main(int argc, char** argv)
{
	struct queueNode *queue = NULL;
	enqueue(&queue, 5);
	enqueue(&queue, 4);
	enqueue(&queue, 3);
	enqueue(&queue, 2);

	printf("%d\n", dequeue(&queue));
	printf("%d\n", dequeue(&queue));
	printf("%d\n", dequeue(&queue));
	printf("%d\n", dequeue(&queue));

	return 0;
}

File: python/queue.py
#!/usr/bin/python


import sys

class QueueNode:
	def __init__(self, data, next=None):
		self.data = data
		self.next = next
		self.last = None

	def __str__(self):
		return str(self.data)


class Queue:
	def __init__(self, data):
		self.queue = QueueNode(data)

	def enqueue(self, data):
		if self.queue == None:
			self.queue = QueueNode(data)
			return
		newQueue = QueueNode(data)

		if self.queue.next == None:
			self.queue.next = newQueue
		else:
			self.queue.last.next = newQueue	
			
		
		self.queue.last = newQueue

	def dequeue(self):
		if(self.queue == None):
			return None
		data = self.queue.data
		self.queue = self.queue.next

		return data


	def front(self):
		return self.queue.data

q = Queue(5)
q.enqueue(4)
q.enqueue(3)
q.enqueue(2)
q.enqueue(1)

print q.dequeue()
print q.dequeue()
print q.dequeue()
print q.dequeue()
print q.dequeue()